src/NativeScript/ObjC/Inheritance/ObjCClassBuilder.mm

```diff
@@ -66,7 +67,7 @@ static void attachDerivedMachinery(GlobalObject* globalObject, Class newKlass, J
         ObjCSuperObject* superObject = ObjCSuperObject::create(vm, superStructure, derivedWrapper, globalObject);
         derivedWrapper->putDirect(vm, vm.propertyNames->superKeyword, superObject, ReadOnly | DontEnum | DontDelete);
 
-        [TNSValueWrapper attachValue:derivedWrapper toHost:instance];
+        globalObject->interop()->objectMap().set(instance, derivedWrapper);
```

Comment:
I've thought about that - the designated initializer on the various adapters adds the object to the map and the `dealloc` method removes it from the map. I can do the same in `ObjCWrapperObject::finishCreation`.